File: available-captures-for-rook/available-captures-for-rook.py
class Solution(object):
def numRookCaptures(self, board):
    output = 0
    for i in range(0,len(board)):
        for j in range(0,len(board[i])):
            if board[i][j] == 'R':
                for r in range(i-1,-1,-1):
                    if board[r][j] == 'B':
                        break
                    if board[r][j] == 'p':
                        output += 1
                        break
                for r in range(i+1,len(board)):
                    if board[r][j] == 'B':
                        break
                    if board[r][j] == 'p':
                        output += 1
                        break
                for c in range(j-1,-1,-1):
                    if board[i][c] == 'B':
                        break
                    if board[i][c] == 'p':
                        output += 1
                        break
                for c in range(j+1,len(board[i])):
                    if board[i][c] == 'B':
                        break
                    if board[i][c] == 'p':
                        output += 1
                        break
    return output
                
                
    

File: available-captures-for-rook/available-captures-for-rook.cpp
class Solution {
public:
    int numRookCaptures(vector<vector<char>>& board) {
        int x = 0, y = 0, output = 0;
        for (int i = 0; i < 8; ++i) {
            for (int j = 0; j < 8; ++j) {
                if (board[i][j] == 'R') {
                    x = i; y = j; break;
                }
            }
        }
        for(int r=x;r >= 0; --r)
        {
            if (board[r][y] == 'B')
                break;
            if (board[r][y] == 'p'){
                ++output;
                break;
            }
        }
        for(int r=x;r < 8; ++r)
        {
            if (board[r][y] == 'B')
                break;
            if (board[r][y] == 'p'){
                ++output;
                break;
            }
        }
        for(int c=y; c >= 0; --c)
        {
            if (board[x][c] == 'B')
                break;
            if (board[x][c] == 'p'){
                ++output;
                break;
            }
        }
        for(int c=y; c < 8; ++c)
        {
            if (board[x][c] == 'B')
                break;
            if (board[x][c] == 'p'){
                ++output;
                break;
            }
        }
        return output;
    }
};
